services/email_verification.js

```diff
@@ -35,45 +41,57 @@ const mailer = require("nodemailer").createTransport({
 });
 
 const server = micro(async req => {
-    const data = await json(req);
+    const data = await json(req.body);
     // generate a token, add it to redis
     let id = idGen.nextId();
-    client.hset(data.user.name, `email_verification`, JSON.stringify({
-        deleteThis: true,
-        id
-    }), async err => {
+    let token = bcrypt.hashSync(`${data.user.email}:${id}`)
+    client.set(`${data.user.email}:email_verify`, token, async err => {
```

Comment:
Client is promisified above but you still use the regular callback functions.

Reply:
Also you don't set a TTL for the key yet, the node_redis repo has an example [here](https://github.com/NodeRedis/node_redis#redis-commands).